apis/rabbitmq/v1beta1/rabbitmquser_types.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	condition "github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// RabbitMQUserPermissions defines permissions for a user on a vhost
+type RabbitMQUserPermissions struct {
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=".*"
+	// Configure - configure permission regex
+	Configure string `json:"configure"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=".*"
+	// Write - write permission regex
+	Write string `json:"write"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=".*"
+	// Read - read permission regex
+	Read string `json:"read"`
+}
```

Comment:
is "" a valid setting value for each of them, or should minLength = 1?

Reply:
"" means deny all, I'll add a comment